### DISCUSSION TITLE
Is it possible to upload an image through a URL instead of through a file upload?

### DISCUSSION THREAD

**Discussion by BrandonOldenhof** [statamic/cms#5887] [Q&A] [answered]
Hello!

We received the following question from one of our clients:

> We're using [Byndr](https://www.bynder.com/nl/) to manage our brand assets and images. Is it possible to upload Byndr assets into the CMS using a URL?

Byndr lets users publish assets to make them available on a specific URL, so I'd like to know if it's possible to upload images by submitting a URL, or if it's possible to add a button to the asset browser UI so I can hook into existing functionality and build it myself.

Thank you!

**Comment by edalzell** [comment]
you want to COPY the byndr image into Statamic? Or just reference it?

**Reply by BrandonOldenhof** [reply]
It would probably be best if it's copied. Referencing would be okay as well, but it would require the referenced image to be shown in de admin panel and the media library as well.
I'm guessing that would be a bit more work.

**Reply by edalzell** [reply]
You could make your own controller that showed a form where they could put an url in, and that controller would fetch the image and make the asset in the correct location?

**Reply by BrandonOldenhof** [reply]
I was thinking about doing that too but I cant find any documentation about how to add that kind of functionality to the asset browser. Do you happen to know where I can find that?

**Comment by jasonvarga** [comment, marked as answer]
Short answer is no. There's nothing built-in that can do that, but it would absolutely be possible to build a custom fieldtype that could support it.

**Reply by BrandonOldenhof** [reply]
Is there any documentation about how to add that kind of functionality to the asset browser?

**Reply by joshuablum** [reply]
Adding that functionality to the asset browser is not (yet) possible, from what I know.

But as Jason and Erin suggested, you can create your own fieldtype and own logic to accomplish this. You can find out more about this in the [docs](https://statamic.dev/extending/fieldtypes)

And you can also open an issue in the https://github.com/statamic/ideas/ repo so it may get added to the core in the future 🙌

**Reply by BrandonOldenhof** [reply]
Thank you for your replies! I'll add it to the [ideas](https://github.com/statamic/ideas/ ) repo :)
